cluster-autoscaler/cloudprovider/oci/nodepools/oci_manager_test.go

```diff
@@ -384,3 +388,70 @@ func TestRemoveInstance(t *testing.T) {
 		}
 	}
 }
+
+func TestNodeGroupFromArg(t *testing.T) {
+	var nodeGroupArg = "clusterId:testClusterId,compartmentId:testCompartmentId,nodepoolTags:ca-managed=true&namespace.foo=bar,min:1,max:5"
+	nodeGroupAutoDiscovery, err := nodeGroupFromArg(nodeGroupArg)
+	if err != nil {
+		t.Errorf("Error: #{err}")
+	}
+	if nodeGroupAutoDiscovery.clusterId != "testClusterId" {
+		t.Errorf("Error: clusterId should be testClusterId")
+	}
+	if nodeGroupAutoDiscovery.compartmentId != "testCompartmentId" {
+		t.Errorf("Error: compartmentId should be testCompartmentId")
+	}
+	if nodeGroupAutoDiscovery.minSize != 1 {
+		t.Errorf("Error: minSize should be 1")
+	}
+	if nodeGroupAutoDiscovery.maxSize != 5 {
+		t.Errorf("Error: maxSize should be 5")
+	}
+	if nodeGroupAutoDiscovery.tags["ca-managed"] != "true" {
+		t.Errorf("Error: ca-managed:true is missing in tags.")
+	}
+	if nodeGroupAutoDiscovery.tags["namespace.foo"] != "bar" {
+		t.Errorf("Error: namespace.foo:bar is missing in tags.")
+	}
+}
+
+func TestValidateNodePoolTags(t *testing.T) {
+
+	var nodeGroupTags map[string]string = nil
+	var nodePoolTags map[string]string = nil
+	var definedTags map[string]map[string]interface{} = nil
+
+	if validateNodepoolTags(nodeGroupTags, nodePoolTags, definedTags) == false {
+		t.Errorf("validateNodepoolTags shouldn't return false for empty tags map")
+	}
+
+	nodeGroupTags = make(map[string]string)
+	nodeGroupTags["test"] = "test"
+
+	if validateNodepoolTags(nodeGroupTags, nodePoolTags, definedTags) == true {
+		t.Errorf("validateNodepoolTags shouldn't return true for tags missing")
+	}
+
+	nodePoolTags = make(map[string]string)
+	nodePoolTags["foo"] = "bar"
+
+	if validateNodepoolTags(nodeGroupTags, nodePoolTags, definedTags) == true {
+		t.Errorf("validateNodepoolTags shouldn't return true for not matching tags")
+	}
```

Comment:
Any reason not to do something like a traditional table driven test like this?

https://go.dev/wiki/TableDrivenTests

My main concern with the current test is that if someone adds a new test case in the middle of this, it will mess up every test that comes after it.

Reply:
I refactored this test to meet the table driven test requirements.
now it runs with test cases given in a map.